tests/Collection.test.ts

```diff
@@ -177,4 +183,127 @@ describe("Collection", () => {
       expect(collection.sourcePath).to.be.null;
     });
   });
+
+  describe("loadFromAmbiguousFile", () => {
+    //// Helper methods ////
+    // Mock createObjectURL and associated fetching
+    const urlToFile: Record<string, File> = {};
+
+    window.URL.createObjectURL = (file: File): string => {
+      console.log(file);
+      const url = "http://mocked-created-url/" + generateUUID();
+      urlToFile[url] = file;
+      return url;
+    };
+
+    window.URL.revokeObjectURL = (url: string): void => {
+      delete urlToFile[url];
+    };
+
+    async function mockFetch(url: string): Promise<Response> {
+      if (url in urlToFile) {
+        const file = urlToFile[url];
+        const fileContents = await new Promise<string>((resolve, reject) => {
+          const reader = new FileReader();
+          reader.onload = () => resolve(reader.result as string);
+          reader.onerror = reject;
+          reader.readAsText(file);
+        });
+        return Promise.resolve(
+          new Response(fileContents, {
+            status: 200,
+            // eslint-disable-next-line
+            headers: { "Content-Type": file.type },
+          })
+        );
+      }
+      return Promise.resolve(
+        new Response("Not Found", {
+          status: 404,
+        })
+      );
```

Comment:
Do these need to be wrapped in `Promise.resolve`s if this is an `async` function? Shouldn't the fact that they're in an async function do that for you?